### DISCUSSION TITLE
New services for `select` & `input_select` entities

### DISCUSSION THREAD

**Discussion by frenck** [frenck/spook#3] [Ideas] [answered]
Looking for some new services to add for `select` entities.

This of course, also applies to `input_select` entities, as they are practically the same (just a different goal/use).

I've added `select.random` & `input_select.random`, which pick a random option from the list.

What else?

**Comment by tdejneka** [comment]
On more than one occasion, I have needed to randomize the order of a list (of colors, TTS phrases, etc). I did it with a Jinja2 template, implementing the Fisher-Yates shuffle algorithm. Ideally `shuffle` would exist as a Jinja2 filter (but it doesn't). 

Not sure if this is feasible but `input_select.shuffle` and `select.shuffle` could be spookily handy. I realize that even if it's feasible, the shuffled entities would revert to their original sort order after a restart/reload (which effectively makes it extra 'not your homie'). ;)

**Reply by frenck** [reply]
`select.shuffle` should really not be done. It is an active entity that relies on e.g., direct communication or polling. Changing that would often not work (and may lead to issues, as it is controlled by an integration)

`input_select.shuffle`, sure 👍 

For `input_select.shuffle` I could update the actual order (in case the helper was created via the UI).

**Reply by frenck** [reply]
added in #26

**Comment by frenck** [comment, marked as answer]
Marking this discussion as resolved. If you have another idea, please create a new discussion.
